From b132289647f8c4f5cfe8814140f8c4889fb5c833 Mon Sep 17 00:00:00 2001
From: Felix Hammarstrand <felix.hammarstrand@flir.se>
Date: Fri, 10 May 2019 14:57:52 +0200
Subject: [PATCH] adapt device tree for ec201 rev b

moved touchscreen settings for ec201 rev a to new ec201-a.dtsi file
new dtb file sherlock-a.dtb for ec201 rev a
use dtb file sherlock.dtb for ec201 rev b and later
---
 arch/arm/boot/dts/Makefile               |  1 +
 arch/arm/boot/dts/imx7ulp-ec201-a.dtsi   | 34 ++++++++++++++++++++++++
 arch/arm/boot/dts/imx7ulp-ec201.dtsi     | 12 ++++-----
 arch/arm/boot/dts/imx7ulp-sherlock-a.dts | 21 +++++++++++++++
 arch/arm/boot/dts/imx7ulp-sherlock.dts   |  2 ++
 5 files changed, 64 insertions(+), 6 deletions(-)
 create mode 100644 arch/arm/boot/dts/imx7ulp-ec201-a.dtsi
 create mode 100644 arch/arm/boot/dts/imx7ulp-sherlock-a.dts

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index bbec7b9c0cbb..c74fe58dab0d 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -727,6 +727,7 @@ dtb-$(CONFIG_SOC_IMX7ULP) += \
 	imx7ulp-14x14-arm2.dtb \
 	imx7ulp-bblc.dtb \
 	imx7ulp-sherlock.dtb \
+	imx7ulp-sherlock-a.dtb \
 	imx7ulp-evk.dtb \
 	imx7ulp-evkb-spi-slave.dtb \
 	imx7ulp-evkb-emmc.dtb \
diff --git a/arch/arm/boot/dts/imx7ulp-ec201-a.dtsi b/arch/arm/boot/dts/imx7ulp-ec201-a.dtsi
new file mode 100644
index 000000000000..3316d4a01dea
--- /dev/null
+++ b/arch/arm/boot/dts/imx7ulp-ec201-a.dtsi
@@ -0,0 +1,34 @@
+
+/*
+ * Copyright 2019 FLIR
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+ 
+/* This device tree is for board ec201 revision a  */
+
+&iomuxc1 {
+
+	pinctrl_lpi2c7: lpi2c7grp {
+		fsl,pins = <
+			IMX7ULP_PAD_PTC12__LPI2C7_SCL   0x27
+			IMX7ULP_PAD_PTC13__LPI2C7_SDA   0x27
+			IMX7ULP_PAD_PTC14__PTC14        0x3 /* PCAPINT_n */
+			IMX7ULP_PAD_PTC7__PTC7          0x3 /* TP_RESET_n */
+			>;
+	};
+};
+
+
+/* Touch screen */
+&lpi2c7 {
+
+	touchscreen:edt_ft5336@38 {
+		interrupt-parent = <&gpio_ptc>;
+		interrupts = <14 GPIO_ACTIVE_HIGH>;
+		reset-gpios = <&gpio_ptc 7 GPIO_ACTIVE_LOW>;
+	};
+};
\ No newline at end of file
diff --git a/arch/arm/boot/dts/imx7ulp-ec201.dtsi b/arch/arm/boot/dts/imx7ulp-ec201.dtsi
index a2a934d6ff9e..b078dcaac787 100644
--- a/arch/arm/boot/dts/imx7ulp-ec201.dtsi
+++ b/arch/arm/boot/dts/imx7ulp-ec201.dtsi
@@ -198,10 +198,10 @@
 
 	pinctrl_lpi2c7: lpi2c7grp {
 		fsl,pins = <
-			IMX7ULP_PAD_PTC12__LPI2C7_SCL   0x27
-			IMX7ULP_PAD_PTC13__LPI2C7_SDA   0x27
-			IMX7ULP_PAD_PTC14__PTC14        0x3 /* PCAPINT_n */
-			IMX7ULP_PAD_PTC7__PTC7          0x3 /* TP_RESET_n */
+			IMX7ULP_PAD_PTE12__LPI2C7_SCL   0x27
+			IMX7ULP_PAD_PTE13__LPI2C7_SDA   0x27
+			IMX7ULP_PAD_PTE14__PTE14        0x3 /* PCAPINT_n */
+			IMX7ULP_PAD_PTF5__PTF5          0x3 /* TP_RESET_n */
 			>;
 	};
 
@@ -472,13 +472,13 @@
 	touchscreen:edt_ft5336@38 {
 		compatible = "edt,edt-ft5306";
 		reg = <0x38>;
-		interrupt-parent = <&gpio_ptc>;
+		interrupt-parent = <&gpio_pte>;
 		interrupts = <14 GPIO_ACTIVE_HIGH>;
 		touchscreen-size-x = <640>;
 		touchscreen-size-y = <480>;
 		touchscreen-inverted-x;
 		touchscreen-inverted-y;
-		reset-gpios = <&gpio_ptc 7 GPIO_ACTIVE_LOW>;
+		reset-gpios = <&gpio_ptf 5 GPIO_ACTIVE_LOW>;
 		status = "okay";
 	};
 };
diff --git a/arch/arm/boot/dts/imx7ulp-sherlock-a.dts b/arch/arm/boot/dts/imx7ulp-sherlock-a.dts
new file mode 100644
index 000000000000..241f58d037a4
--- /dev/null
+++ b/arch/arm/boot/dts/imx7ulp-sherlock-a.dts
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2019 FLIR
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+/dts-v1/;
+
+/* This device tree is for board ec201 revision a  */
+
+#include <dt-bindings/input/input.h>
+#include "imx7ulp.dtsi"
+#include "imx7ulp-ec201.dtsi"
+#include "imx7ulp-ec201-a.dtsi"
+
+/ {
+	model = "FLIR i.MX7ULP Sherlock rev a board";
+	compatible = "flir,imx7ulp-sherlock", "flir,imx7ulp-ec201-a","flir,imx7ulp-ec201", "fsl,imx7ulp", "Generic DT based system";
+};
diff --git a/arch/arm/boot/dts/imx7ulp-sherlock.dts b/arch/arm/boot/dts/imx7ulp-sherlock.dts
index 906d679e84b2..b9e1041b8b00 100644
--- a/arch/arm/boot/dts/imx7ulp-sherlock.dts
+++ b/arch/arm/boot/dts/imx7ulp-sherlock.dts
@@ -8,6 +8,8 @@
 
 /dts-v1/;
 
+/*This device tree is for latest version of ec201*/
+
 #include <dt-bindings/input/input.h>
 #include "imx7ulp.dtsi"
 #include "imx7ulp-ec201.dtsi"
-- 
2.17.1

